ui/src/app/pages/workspace/create-billing-account-modal.tsx

```diff
@@ -1,76 +1,407 @@
+const BillingConfirmItem = ({title, value, dataTestId}) => {
+  return <FlexRow id = {`${dataTestId}-wrapper`} style={{marginTop: '5px'}}>
+    <div style={{width: '170px'}}>{title}:</div>
+    <div data-test-id={dataTestId}>{value}</div>
+  </FlexRow>;
+};
 
-        <hr style={{width: '100%', backgroundColor: colors.primary, borderWidth: '0px', height: '1px',
-          marginTop: '1rem',
-          marginBottom: '0.5rem'}}/>
+export const CreateBillingAccountModal = ({onClose}: Props) => {
+  const {profile: {
+    contactEmail,
+    givenName,
+    familyName,
+    verifiedInstitutionalAffiliation,
+    username
+  }} = useStore(profileStore);
+  const [currentStep, setCurrentStep] = useState(0);
+  const [phoneNumber, setPhoneNumber] = useState();
+  const [invalidPhoneNumberInput, setInvalidPhoneNumberInput] = useState<boolean>(null);
+  const [useCreditCard, setUseCreditCard] = useState<boolean>(null);
+  const [nihFunded, setNihFunded] = useState<boolean>(null);
 
-        <ModalFooter style={{marginTop: 0, justifyContent: 'flex-start'}}>
-          <FlexColumn>
-            <TextColumn>
-              <p style={{marginTop: 0}}>Don't already have a Google Cloud account?</p>
-              <a href='https://cloud.google.com' target='_blank'>Create an account</a>
-            </TextColumn>
+  const validatePhoneNumber = (phoneInput: string) => {
+    const sanitizedPhone = phoneInput.replace(/\D/g, '');
 
-            <TextColumn>
-              <div>Add [ {this.props.profileState.profile.username} ] as a 'User' to your Google Cloud billing account.</div>
-              <a href='https://console.cloud.google.com/billing' target='_blank'>Add user</a>
-            </TextColumn>
-          </FlexColumn>
-        </ModalFooter>
+    if (sanitizedPhone.length >= 10 && sanitizedPhone.length <= 11) {
+      setInvalidPhoneNumberInput(false);
+      setPhoneNumber(sanitizedPhone);
+    } else {
+      setInvalidPhoneNumberInput(true);
+    }
+  };
 
-        <FontAwesomeIcon
-          icon={faTimes}
-          size='lg'
-          style={{
-            color: colors.accent,
-            position: 'absolute',
-            top: '1rem',
-            right: '1rem',
-            cursor: 'pointer'
-          }}
-          onClick={() => this.props.onClose()}
-        />
-      </Modal>;
+  const sendCreateBillingEmail = fp.flow(
+    withErrorModal({
+      title: 'Failed To Send Email',
+      message: 'An error occurred trying to send email. Please try again.',
+    })
+  )(async() => {await profileApi().sendBillingSetupEmail(
+    {
+      phone: phoneNumber,
+      paymentMethod: useCreditCard ? BillingPaymentMethod.CREDITCARD : BillingPaymentMethod.PURCHASEORDER,
+      isNihFunded: nihFunded,
+      institution: verifiedInstitutionalAffiliation.institutionDisplayName
     }
-  }
-);
+  );
+    setCurrentStep(4);
+  });
+
+  return <Modal width={650} onRequestClose={() => onClose()}>
+        <ModalBody style={{marginTop: '-4'}}>
```

Comment:
Negative margins should be considered a hacky last resort. This one, I actually don't see where it takes effect int he app. Can it be removed?